src/controllers/UserController.js

```diff
@@ -15,15 +17,31 @@ class UserController {
         };
 
         const data = {
-            ...req.body,
             location,
-        };
+            ...req.body,
+        }
+
+        if(password.length < 8) {
+            res.status(400).json({ error: 'Senha inválida' });
+            next();
+        }
 
         try {
+            // Cria o usuário no miaAjuda
             const result = await this.userService.createUser(data);
+
+            // Cria o usuário no firebase
+            await firebase.auth().createUser({
+                email: req.body.email,
+                password: req.body.password,
+                displayName: req.body.name,
+                phoneNumber: req.body.phone
+            });
+
             res.status(201).json(result);
             next();
         } catch (err) {
+            await this.userService.removeUser(email);
             res.status(400).json({ error: err });
             next();
         }
```

Comment:
Nessa situação, qualquer que seja a exceção gerada, entrará no _catch_ e excluirá o usuário. Isto é, se alguém passar um atributo que é único no banco, um e-mail por exemplo, caso já exista esse e-mail no banco, gerará exceção e excluirá o usuário da base de dados. Uma solução é realizar a exclusão apenas quando o erro for ocasionado pelo firebase.